contributors/design-proposals/live-and-inplace-vertical-scaling.md

```diff
@@ -0,0 +1,325 @@
+    ResizeRequest ResizeRequest
+    ...
+}
+```
+
+ResizeRequest has two variables, RequestStatus and NewResources.
+RequestStatus represents the status of a resource resizing request.
+ResizeRequested indicates resource resizing for a pod is requested.
+ResizedAccepted and ResizedRejected means that the requested resource resizing is accepted and rejected, respectively, by the Scheduler.
+The NewResources is an array indexed by a container’s index and its each entry holds new resource requirements of a container that needs to resize.
+
+Given a new PodSpec with new resource requirements from a client, first the 'API server' validates it.
+If it is valid, the 'API server' sets the RequestStatus to ResizedRequested and copies the new resource requirements of each container into the NewResources.
+Also, the 'API server' restores the resource requirements of each container of the PodSpec to the original and writes the revised PodSpec to ETCD to communicate with the Scheduler.
+This is because at this moment the PodSpec on ETCD shouldn’t be updated with new resource requirements.
+
+For a pod with ResizeRequested, the 'Scheduler' checks if the node on which the pod currently runs has enough resources to resize the pod.
+The Scheduler notify the 'API server' of the result via 'Resizing' API operation, which will be describe below.
+
+* **Resizing**
+
+A new API, Resizing, for the 'scheduler' is introduced:
+
+```go
+// Resizing resizes the resources allocated to a pod
+type Resizing struct {
+    metav1.TypeMeta
+    metav1.ObjectMeta
+    Request ResizeRequest
+}
+```
+
+Resizing has the metadata of a pod to resize and a value of ResizeRequest that holds the status of a resizing request, which indicates whether the resizing is feasible or not, and new resource requirements of the pod.
+
+Once the 'Scheduler' determine whether a resource resizing on a pod is feasible, or not, it notifies to the API server via this Resizing API.
+ObjectMeta.Namespace/Name are set to the Namespace and Name of the pod to resize and ResizeRequest is set with new resource requirements.
+
+Given a Resizing API operation, the ResizeStatus of the PodSpec of a Pod is updated according to that of the Resizing operation.
+If the ResizeStatus is ResizeAccepted, the API server updates the ResourceRequirement of each container of a pod with new resource requirements on ETCD.
+
+* **PodResized**
+
+A new pod condition, PodResized, and condition status for that is added to v1.PodCondition:
+
+```go
+// These are valid conditions of pod.
+const (
+       // PodResized represents the status of the resizing process for this pod
+    PodResized PodConditionType = "PodResized"
+    PodReasonUnresizable   = "Unresizable"
+    PodReasonResizerFailed = "ResizerFailed"
+)
+const (
+    ConditionRequested ConditionStatus = "Requested"
+    ConditionAccepted  ConditionStatus = "Accepted"
+    ConditionRejected  ConditionStatus = "Rejected"
+    ConditionDone      ConditionStatus = "Done"
```

Comment:
How long the pod status is kept?

Reply:
The ConditionRequest and ConditionAccepted are intermediate states. So, if the scheduler and kubelet are operational, the pod status will be changed to ConditionRejected or ConditionDone. Once the pod status becomes ConditionRejected or ConditionDone, it won't be changed unless a new resizing request comes in.